### PR TITLE
exit app on xapp error

### PR DESCRIPTION
**Pull request by mbilokonsky**
Just followed the example from the npm page for artsy-xapp - hopefully this'll catch 502s.

### CODE REVIEW

**Comment by orta** [comment]
Yeah, I agree with this - merge on green

File: src/index.js [hunk 1]
```diff
@@ -30,6 +30,10 @@ setup(app)
 
 // Connect to Redis
 cache.setup(() => {
+  // if we can't get an xapp token, just exit and let the whole system try
+  // again - this prevents a sustained broken state when gravity returns a
+  // 502 during force startup.
+  artsyXapp.on('error', process.exit)
```
Comment (sweir27) [inline src/index.js]: Would it be possible to log some kind of helpful message if this happens? Just ran into a similar `.env` issue with @zephraph and it correctly exited but was a little hard to figure out why.

Comment (zephraph) [reply src/index.js]: yessssss... Hopefully that spits out an error message that we could console.log or something.

Comment (orta) [reply src/index.js]: I'll add this to https://github.com/artsy/force/pull/2829

**Comment by mbilokonsky** [comment]
Thanks Orta!

**Pull request merged**